tests/unit/test_allocator.cpp

```diff
@@ -331,7 +330,12 @@ TEST_F(AllocatorTest, testIncomingEdgesSet) {
             hnswIndex->getHNSWIndex()->size_links_per_element_ * vec_max_level + 1 +
             AllocatorTest::vecsimAllocationOverhead;
     }
-    expected_allocation_delta += AllocatorTest::setNodeSize;
+
+    // Expect that the first element is pushed to the incoming edges vector of element 1 in level 0.
+    // Then, we account for the capacity of the buffer that is allocated for the vector data.
+    expected_allocation_delta += hnswIndex->getHNSWIndex()->getIncomingEdgesPtr(1, 0)->capacity() *
```

Comment:
Why don't you create with capacity 0?

Reply:
That's just how STL works. When a new (empty) vector is created, the data buffer is not allocated at all